### DISCUSSION TITLE
Slowdown by vcpkg msbuild integration

### DISCUSSION THREAD

**Discussion by autoantwort** [microsoft/vcpkg#19406] [Q&A]
When we use vcpkg for our projects (solutions says 108 projects) our pipeline runs ~6 minutes longer (39 -> 45 minutes) while the vcpkg.json file is "empty" (no dependencies). Imo 6 minutes for doing nothing is quite long. It seems that vcpkg is called in 103 Projects, so it needs 3,5 Seconds per projects. When looking at the logs it seems that most of the time it waits for the file lock (~2.5 - 3 seconds), but the reported total elapsed time is ~15.9 us ... (#16850). So when the file is locked another tool must run at the same time. Why needs vcpkg so long to run and is there a way to reduce the file locked time?

**Comment by PhoebeHui** [comment]
I have no idea about this, @vicroms, could you please help take a look?

**Comment by meastp** [comment]
This hurts us as well. IIRC, @ras0219-msft wrote something about this on Slack the other day. I think concurrent build is disabled when using manifest mode because of the file lock...?

**Comment by pfaco** [comment]
Is there any known workaround for this? This is the only downside since our adoption of vcpkg.

**Comment by ras0219** [comment]
For CMake, you can explicitly disable the manifest install step via `-DVCPKG_MANIFEST_INSTALL=OFF` (https://github.com/microsoft/vcpkg/blob/master/docs/users/manifests.md#vcpkg_manifest_install).

As a workaround, could you try:
```
file(TIMESTAMP "${CMAKE_CURRENT_LIST_DIR}/vcpkg.json" vcpkg_json_timestamp)
if(VCPKG_JSON_TIMESTAMP_PREV STREQUAL vcpkg_json_timestamp)
    set(VCPKG_MANIFEST_INSTALL OFF CACHE STRING "" FORCE)
else()
    set(VCPKG_MANIFEST_INSTALL ON CACHE STRING "" FORCE)
endif()

project(...)
set(VCPKG_JSON_TIMESTAMP_PREV "${vcpkg_json_timestamp}" CACHE STRING "" FORCE)
```

This should only run the install when the timestamp changes. If you try this out, please let us know if this workaround solves the problem to your satisfaction :)

**Reply by autoantwort** [reply]
Does it make sense to make this an official feature and option?

**Reply by Osyotr** [reply]
@autoantwort absolutely. Current behavior slows down configure step significantly, especially on WSL. And VS runs configure every time any CMakeLists.txt changed...

**Comment by pfaco** [comment]
thanks @ras0219 it solved the problem here.
